### DISCUSSION TITLE
Minio creating excessive directories.

### DISCUSSION THREAD

**Discussion by Jazz9** [minio/minio#17673] [Q&A] [answered]
I have been monitoring our minio cluster and have found that the Prometheus metrics show many directories being deleted and created.

It doesn't seem to corelate to s3 activity directly, is it the scanner? avg of 50 per second per drive

minio.RELEASE.2023-06-29T05-12-28Z
6 x nodes with 11 HDD each

[image](https://github.com/minio/minio/assets/39378577/5abcbd0d-2bac-4f4b-b4e4-db12e8fa3bb2)

![image](https://github.com/minio/minio/assets/39378577/1554a698-d198-4f17-aeba-e25502ea355d)

**Comment by Jazz9** [comment]
Interesting - it is caused by a deletion of objects from veeam. The rate limiting doesn't help as the requests are multideletes. 

Any ideas how to rate limit or improve the performance of delete operations?

**Reply by vadmeste** [reply]
@Jazz9 what is the original issue ? what do you want to do a rate limit in the first place ?

**Reply by Jazz9** [reply]
The deletes caused the system to become slow for users. HDD ran out of iops, 20+ queue depth. There is a rate limiter set in minio at 50 requests but with multi-deletes it threads out after the limiter.

**Comment by harshavardhana** [comment, marked as answer]
> Interesting - it is caused by a deletion of objects from veeam. The rate limiting doesn't help as the requests are multideletes.

Veeam workload deployment must use NVMe or SSDs- if you use HDDs things will slow down and there is nothing that can be done from the IOPs point of view.
